v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/FileFormatFactory.java

```
@@ -0,0 +1,124 @@
+         */
+        switch (outputFileFormat) {
+            case TEXT:
+                output =
+                        records.apply(
+                                "Write Text File(s)",
+                                WriteToGCSText.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .withTempLocation(options().getTempLocation())
+                                        .build());
+                break;
+
+            case AVRO:
+                output =
+                        records.apply(
+                                "Write Avro File(s)",
+                                WriteToGCSAvro.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .withTempLocation(options().getTempLocation())
+                                        .build());
+                break;
+
+            case PARQUET:
+                output =
+                        records.apply(
+                                "Write Parquet File(s)",
+                                WriteToGCSParquet.newBuilder()
+                                        .withOutputDirectory(options().getOutputDirectory())
+                                        .withOutputFilenamePrefix((options().getOutputFilenamePrefix()))
+                                        .withNumShards(options().getNumShards())
+                                        .build());
+                break;
+
+            default:
+                break;
```

what happens when you get to this line?  "output" will be null which we will return.

---

Yes. And didn't add any Exceptions since already we are checking the file format before entering the switch case.

---

LOG.info(errorMessage);
throw new IllegalArgumentException(errorMessage);

Added these statements.

---

We would still want to throw an IllegalArgumentException or UnsupportedOperationException in the default case. The corner case here is that if a new file format is added to the FileFormat enum but is not added here due to oversight.